### ISSUE TITLE
support `csv` filter

### ISSUE THREAD

**Issue by vladimir-dd** [timberio/vector#7979]
`csv(headers[, separator[, quotingcharacter]])`

**Comment by jszwedko** [comment]
Just to make sure I understand these correctly, we call these parsers, right? We actually have a `parse_csv` function in VRL already, but it lacks support for setting the headers.

**Comment by vladimir-dd** [comment]
yes, I actually wanted to update our `parse_csv` VRL function and then just re-use it for my grok filter

**Comment by jszwedko** [comment]
Ah, gotcha, these are integrated into the grok patterns. 👍
